synapse/rest/client/keys.py

```diff
@@ -375,9 +376,19 @@ async def on_POST(self, request: SynapseRequest) -> Tuple[int, JsonDict]:
         user_id = requester.user.to_string()
         body = parse_json_object_from_request(request)
 
-        if self.hs.config.experimental.msc3967_enabled:
+        if (
+            self.hs.config.experimental.msc3967_enabled
+            or self.hs.config.experimental.msc3861.enabled
+        ):
```

Comment:
Should we assert in the config that `msc3967_enabled` must be `true` if `msc3861` is enabled?

Reply:
The comment should make it clearer now: a39c3d3f2

Reply:
The refactoring makes it a lot clearer, thanks!